open-build-service-api/src/lib.rs

```diff
@@ -161,15 +162,81 @@ pub struct DirectoryEntry {
 #[derive(Deserialize, Debug)]
 pub struct Directory {
     pub name: String,
-    pub rev: String,
-    pub vrev: String,
+    pub rev: Option<String>,
+    pub vrev: Option<String>,
     pub srcmd5: String,
-    #[serde(rename = "entry")]
+    #[serde(default, rename = "entry")]
     pub entries: Vec<DirectoryEntry>,
     #[serde(default, rename = "linkinfo")]
     pub linkinfo: Vec<LinkInfo>,
 }
 
+#[derive(Clone, Deserialize, Serialize, Debug)]
+pub struct CommitEntry {
+    pub name: String,
+    pub md5: String,
```

Comment:
Is this a hexdigest? a base64? misc ? :) would be nice to at least document it or have a newtype wrapper for convenience/validation. Can be later though